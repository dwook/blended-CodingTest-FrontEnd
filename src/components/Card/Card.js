import React from 'react';
import { Link } from 'react-router-dom';
import './Card.scss';
import { BASE_URL_POSTSER } from '../../constants';

function Card({ poster_path, title, id }) {
  return (
    <div className="card">
      <Link to={`/movies/${id}`}>
        <div className="thumbnail">
          {poster_path ? (
            <img
              src={BASE_URL_POSTSER + poster_path}
              alt={title}
            />
          ) : (
            <p>No Image</p>
          )}
        </div>
        <div className="info">
          <div className="title">{title}</div>
          <div></div>
        </div>
      </Link>
    </div>
  );
}

export default Card;
