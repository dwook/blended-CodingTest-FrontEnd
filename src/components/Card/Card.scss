.card {
  width: 20%;
  padding: 10px;
  box-sizing: border-box;
  position: relative;
  &:hover {
    .thumbnail,
    .info {
      transform: scale(1.1);
      -webkit-transition: 0.5s ease;
      transition: 0.5s ease;
    }
  }
  .thumbnail {
    height: 90%;
    position: relative;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    background: rgba(255, 255, 255, 0.2);
    color: #212529;
    &:after {
      content: '';
      display: block;
      width: 100%;
      height: 50%;
      position: absolute;
      bottom: 0;
      background: rgb(33, 37, 41);
      background: linear-gradient(
        0deg,
        rgba(33, 37, 41, 1) 0%,
        rgba(113, 161, 255, 0) 100%
      );
    }
    img {
      width: 100%;
    }
  }
  .info {
    position: absolute;
    width: 90%;
    top: 70%;
    padding: 0 20px;
    box-sizing: border-box;
    color: #fff;
    .title {
      font-size: 20px;
      line-height: 1.4em;
      height: 2.8em;
      display: -webkit-box;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;
      overflow: hidden;
    }
  }
}
