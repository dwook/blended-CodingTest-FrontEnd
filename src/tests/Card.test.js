import React from 'react';
import Card from '../components/Card/Card';
import { shallow } from 'enzyme';

describe('<Card />', () => {
  let wrapper1;
  let wrapper2;
  const movie_with_poster = {
    adult: false,
    backdrop_path: '/6vKPprzJPUSDW3IyJiZ2RyrpChY.jpg',
    belongs_to_collection: null,
    budget: 0,
    genres: [
      {
        id: 28,
        name: 'Action'
      },
      {
        id: 35,
        name: 'Comedy'
      },
      {
        id: 10751,
        name: 'Family'
      }
    ],
    homepage: '',
    id: 656216,
    imdb_id: 'tt9149946',
    original_language: 'en',
    original_title: 'Bernie the Dolphin 2',
    overview:
      "The kids are thrilled that Bernie has come back. But so has their old enemy Winston, who's about to kidnap the talented dolphin. Kevin and Holly must rescue their splashy friend before it's too late.",
    popularity: 0,
    poster_path: '/r0U3mBQEOJ6kKpxZ0F8WFfjCldK.jpg',
    production_companies: [],
    production_countries: [],
    release_date: '2019-12-17',
    revenue: 0,
    runtime: 99,
    spoken_languages: [],
    status: 'Released',
    tagline: '',
    title: 'Bernie the Dolphin 2',
    video: false,
    vote_average: 6,
    vote_count: 1
  };

  const movie_without_poster = {
    adult: false,
    backdrop_path: '/6vKPprzJPUSDW3IyJiZ2RyrpChY.jpg',
    belongs_to_collection: null,
    budget: 0,
    genres: [
      {
        id: 28,
        name: 'Action'
      },
      {
        id: 35,
        name: 'Comedy'
      },
      {
        id: 10751,
        name: 'Family'
      }
    ],
    homepage: '',
    id: 656216,
    imdb_id: 'tt9149946',
    original_language: 'en',
    original_title: 'Bernie the Dolphin 2',
    overview:
      "The kids are thrilled that Bernie has come back. But so has their old enemy Winston, who's about to kidnap the talented dolphin. Kevin and Holly must rescue their splashy friend before it's too late.",
    popularity: 0,
    poster_path: null,
    production_companies: [],
    production_countries: [],
    release_date: '2019-12-17',
    revenue: 0,
    runtime: 99,
    spoken_languages: [],
    status: 'Released',
    tagline: '',
    title: 'Bernie the Dolphin 2',
    video: false,
    vote_average: 6,
    vote_count: 1
  };

  beforeEach(() => {
    wrapper1 = shallow(<Card {...movie_with_poster} />);
    wrapper2 = shallow(<Card {...movie_without_poster} />);
  });

  it('matches snapshot', () => {
    expect(wrapper1).toMatchSnapshot();
  });

  it('must have title', function() {
    expect(
      wrapper1.contains(<div className="title">Bernie the Dolphin 2</div>)
    ).toBe(true);
  });

  it('must show an image if it has a poster path', function() {
    expect(wrapper1.find('img')).toHaveLength(1);
  });

  it('must show no image text if it does not have a poster path', function() {
    expect(wrapper2.contains(<p>No Image</p>)).toBe(true);
  });
});
