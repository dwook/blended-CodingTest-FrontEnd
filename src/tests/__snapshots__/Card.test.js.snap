// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot 1`] = `
<div
  className="card"
>
  <Link
    to="/movies/656216"
  >
    <div
      className="thumbnail"
    >
      <img
        alt="Bernie the Dolphin 2"
        src="https://image.tmdb.org/t/p/w500/r0U3mBQEOJ6kKpxZ0F8WFfjCldK.jpg"
      />
    </div>
    <div
      className="info"
    >
      <div
        className="title"
      >
        Bernie the Dolphin 2
      </div>
      <div />
    </div>
  </Link>
</div>
`;
